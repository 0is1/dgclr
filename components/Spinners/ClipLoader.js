import React from 'react';
import { ClipLoader } from 'react-spinners';
import { Box, Flex } from 'rebass';
import colors from 'components/colors';

const Spinner = () => (
  <Flex alignItems="center" p="5rem 0" justifyContent="center">
    <Box>
      <ClipLoader className="loading" sizeUnit="px" size={80} color={colors.info} loading />
    </Box>
  </Flex>
);

export default Spinner;
