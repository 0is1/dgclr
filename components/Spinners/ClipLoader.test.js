/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Box } from 'rebass';
import { ClipLoader as ClipLoaderSpinner } from 'react-spinners';
import ClipLoader from './ClipLoader';

describe('Layout component', () => {
  it('Match snapshot', () => {
    const component = renderer.create(<ClipLoader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders correctly', () => {
    const enzymeWrapper = shallow(<ClipLoader />);
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(Box).length).toEqual(1);
    expect(enzymeWrapper.find(ClipLoaderSpinner).length).toEqual(1);
  });
});
