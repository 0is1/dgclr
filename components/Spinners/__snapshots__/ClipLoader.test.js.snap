// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Match snapshot 1`] = `
<div
  className="sc-bwzfXH dbiuev sc-bdVaJa enBfDC"
>
  <div
    className="sc-bdVaJa iHZvIS"
  >
    <div
      className="css-1f05wiv-style-style"
    />
  </div>
</div>
`;
